test/deployframework/framework.go

```diff
@@ -0,0 +1,335 @@
+// needed to customize a ReportingFramework object
+type ReportingFrameworkConfig struct {
+	UseKubeProxyForReportingAPI bool
+	UseRouteForReportingAPI     bool
+	HTTPSAPI                    bool
+	Namespace                   string
+	KubeConfigPath              string
+	ReportingAPIURL             string
+	RouteBearerToken            string
+	ReportResultsOutputPath     string
+}
+
+// New is the constructor function that creates and returns a new DeployFramework object
+func New(
+	logger logrus.FieldLogger,
+	nsPrefix,
+	manifestDir,
+	kubeconfig,
+	cleanupScriptPath,
+	loggingPath,
+	reportingOperatorImageRepo,
+	reportingOperatorImageTag string,
+) (*DeployFramework, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build a kube config from %s: %v", kubeconfig, err)
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize the k8s clientset: %v", err)
+	}
+
+	apiextClient, err := apiextclientv1beta1.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize the apiextensions clientset: %v", err)
+	}
+
+	meteringClient, err := meteringv1.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize the metering clientset: %v", err)
+	}
+
+	deployFramework := &DeployFramework{
+		NamespacePrefix:            nsPrefix,
+		ManifestsDir:               manifestDir,
+		CleanupScriptPath:          cleanupScriptPath,
+		KubeConfigPath:             kubeconfig,
+		LoggingPath:                loggingPath,
+		ReportingOperatorImageRepo: reportingOperatorImageRepo,
+		ReportingOperatorImageTag:  reportingOperatorImageTag,
+		Logger:                     logger,
+		Client:                     client,
+		APIExtClient:               apiextClient,
+		MeteringClient:             meteringClient,
+		Config:                     ReportingFrameworkConfig{},
+	}
+
+	return deployFramework, nil
+}
+
+// Setup handles the process of deploying metering, and waiting for all necessary resources
+// to become ready in order to proceed with running the reporting tests.
+func (df *DeployFramework) Setup(cfg deploy.Config, namespace, testOutputPath string, targetPods int) (*ReportingFrameworkConfig, error) {
+	var err error
+
+	cfg.OperatorResources, err = deploy.ReadMeteringAnsibleOperatorManifests(df.ManifestsDir, cfg.Platform)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize objects from manifests: %v", err)
+	}
+
+	// update the omitted namespace fields in the test table
+	df.Config.Namespace = namespace
+	cfg.Namespace = namespace
+	cfg.MeteringConfig.ObjectMeta = meta.ObjectMeta{
+		Name:      meteringconfigMetadataName,
+		Namespace: namespace,
+	}
+
+	df.Logger.Debugf("Deployer config: %+v", cfg)
+
+	df.Deployer, err = deploy.NewDeployer(cfg, df.Logger, df.Client, df.APIExtClient, df.MeteringClient)
```

Comment:
The only issue I see with my comment from above is how we keep track of the `Deployer` object and use it between Install and Uninstall. This is the reason I think we may need another type of object which is returned by `Setup()` which you could call Teardown on, this way `Deployer` isn't a field on `DeployFramework` since that's more of an intermediate object/state.

For example:
```
df := deployFramework.New(...)
deployCtrl, err := df.Setup()
if err != nil { ... }
defer deployCtrl.Teardown()
reportingFramework := deployCtrl.ReportingFramework
runReportTests(reportingFramework)
```